services/spar/src/Spar/Run.hs

```diff
@@ -74,12 +74,19 @@ initCassandra opts lgr = do
 -- this would create the "Listening on..." log message there, but it may also have other benefits.
 runServer :: Opts -> IO ()
 runServer sparCtxOpts = do
-  sparCtxLogger <- Log.mkLogger (toLevel $ saml sparCtxOpts ^. SAML.cfgLogLevel) (logNetStrings sparCtxOpts)
-  mx <- metrics
-  sparCtxCas <- initCassandra sparCtxOpts sparCtxLogger
   let settings = Warp.defaultSettings & Warp.setHost (fromString shost) . Warp.setPort sport
       shost :: String = sparCtxOpts ^. to saml . SAML.cfgSPHost
       sport :: Int    = sparCtxOpts ^. to saml . SAML.cfgSPPort
+  (wrappedApp, sparCtxLogger -> logger) <- mkApp sparCtxOpts
```

Comment:
I'm often a fan of view patterns, but I don't love them in the binding position of do blocks, I know @jschaul tends to dislike this sort of thing also; 

I prefer fmap in these cases; what do you think of:

- `(wrappedApp, logger) <- over _2 sparCtxLogger <$> mkApp sparCtxOpts`
- `(wrappedApp, logger) <- second sparCtxLogger <$> mkApp sparCtxOpts`

Or the arguably much clearer:

```haskell
(wrappedApp, ctxOpts) <- mkApp sparCtxOpts
let logger = sparCtxLogger ctxOpts
```

Reply:
I don't think your alternatives are much clearer.  But I will change it to the last option and try to change my habits in the future.